src/components/templates/ScreenShare/ScreenShare.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useState } from "react";
+import AgoraRTC, { IAgoraRTCClient } from "agora-rtc-sdk-ng";
+import useAgoraScreenShare from "./agoraHooks/useAgoraScreenShare";
+import Player from "./components/Player/Player";
+import useAgoraCamera from "./agoraHooks/useAgoraCamera";
+import "./ScreenShare.scss";
+import useAgoraRemotes from "./agoraHooks/useAgoraRemotes";
+import Audience from "./components/Audience/Audience";
+import Button from "./components/Button/Button";
+import LeaveStageModal from "./components/LeaveStageModal/LeaveStageModal";
+
+const remotesClient: IAgoraRTCClient = AgoraRTC.createClient({
+  codec: "h264",
+  mode: "rtc",
+});
+
+const screenClient: IAgoraRTCClient = AgoraRTC.createClient({
+  codec: "h264",
+  mode: "rtc",
+});
+
+const cameraClient: IAgoraRTCClient = AgoraRTC.createClient({
+  codec: "h264",
+  mode: "rtc",
+});
```

Comment:
These also look like they should be `useMemo`'d or similar as well?

Reply:
They are just consts, any reason to use `useMemo`?

Reply:
Based on the call to .createClient it would seem that they are constructing stage full objects, so every single time the component function is called to check if it needs to render (which happens a lot in react) they would be re-created. I haven’t looked deeply into the code for what it does, but that seems suboptimal.